src/openrct2/world/footpath.c

```diff
@@ -493,7 +505,9 @@ void game_command_place_footpath(sint32 *eax, sint32 *ebx, sint32 *ecx, sint32 *
         *edx & 0xFF,
         (*ebx >> 8) & 0xFF,
         *ebx & 0xFF,
-        *edi & 0xFF
+        *edi & 0xFF,
+        (*ebp >> 8) & 0xFF,
```

Comment:
Mask with the flag only and then shift. It will make the intentions here much clearer